libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod error;
+mod scar;
+mod submit;
+
+use sawtooth_sdk::messages::batch::BatchList;
+
+pub use error::Error;
+pub use scar::{SmartContract, SmartContractMetadata};
+use submit::{submit_batches, wait_for_batches};
+
+/// A client that can be used to submit transactions to a scabbard service.
+pub struct ScabbardClient {
+    url: String,
+}
+
+impl ScabbardClient {
+    /// Create a new `ScabbardClient` with the given `url`.
+    pub fn new(url: &str) -> Self {
+        Self { url: url.into() }
+    }
+
+    /// Submit the given batches to the scabbard service. Optionally wait the given number of
+    /// seconds for batches to commit.
+    pub fn submit(&self, batches: BatchList, wait: Option<u64>) -> Result<(), Error> {
+        let batch_link = submit_batches(&self.url, batches)?;
+        if let Some(wait_secs) = wait {
+            wait_for_batches(&batch_link, wait_secs)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Load a `SmartContract` from a .scar file on the local filesystem.
+    ///
+    /// If the argument is a file path (contains a '/'), this will attempt to load the .scar from
+    /// the specified location. If the argument is not a file path, this will attempt to load the
+    /// .scar from the directories listed in the SCAR_PATH environment variable. When loading from
+    /// a directory in SCAR_PATH, the '.scar' file extension is optional.
+    pub fn load_scar(scar: &str) -> Result<SmartContract, Error> {
```

Comment:
This should probably be a function outside of this interface.

Reply:
Removed this method